packages/workers-shared/utils/types.ts

```diff
@@ -20,6 +20,13 @@ export const RouterConfigSchema = z.object({
 	...InternalConfigSchema.shape,
 });
 
+export const EyeballRouterConfigSchema = z.union([
+	z.object({
+		limitedAssetsOnly: z.boolean().optional(),
+	}),
+	z.null(),
+]);
```

Comment:
Why support both `undefined` and `null`. I don't believe there is any semantic difference here? Can we just drop the `null` and simplify this and the `RequiredEyeballRouterConfig` type?

Reply:
the optional param binding which provides this param (as the EYEBALL_CONFIG binding) provides the json when present, otherwise the value is presented in the environment as null, so that union reflects that implementation

the field limitedAssetsOnly within the object is optional to be defensive, but currently the code providing the object always sets the field when providing the object. this can be changed, but i prefer being defensive, as this config may be extended and not all fields may be set in the future